openmc/checkvalue.py

```diff
@@ -1,3 +1,17 @@
+from collections import Iterable
+from numbers import Integral
+
+import numpy as np
+
+def _isinstance(value, expected_type):
```

Comment:
Personally, I'm fine with this extension to support Numpy <v1.9, but would want @paulromano's opinion before permitting this into the codebase. My only hesitation is that this routine *may* greatly impact the performance of the API. The performance doesn't really matter for the input generation since it's so ridiculously fast, but I have observed a hit when using these sorts of error checking routines for post-processing large tally datasets. Albeit, I made those observations for some of my post-processing code built atop the OpenMC Python API, but the point is still the same - in general, we need to be prudent about the tradeoff between user-centric error handling and performance. I would keep this new routine, but leave a comment that this can be removed in a year or two once NumPy v1.9 is standard everywhere.